code/game/machinery/_machinery.dm

```diff
@@ -222,6 +222,10 @@ Class Procs:
 		return FALSE
 	else if(silicon && !(interaction_flags_machine & INTERACT_MACHINE_ALLOW_SILICON))
 		return FALSE
+	if(!silicon && !Adjacent(user))
```

Comment:
you shouldn't need to check the silicon value 3 times in a row here, restructure the ifs a little